### PR TITLE
Provide version header.

### PR DESCRIPTION
**Pull request by odashi**
This PR provides:
- The version header `primitiv/version.h` which is automatically generated by CMake configurations.
- Additional scripts in `FindPrimitiv.cmake` to check the installed version.

### CODE REVIEW

File: cmake/FindPrimitiv.cmake [hunk 1]
```diff
@@ -77,12 +81,58 @@ find_library(PRIMITIV_C_LIBRARIES NAMES primitiv_c
...
 if (PRIMITIV_C_INCLUDE_DIR AND PRIMITIV_C_LIBRARIES)
-  message("-- Found Primitiv C API: " ${PRIMITIV_C_LIBRARIES})
-  #message("-- Primitiv C API include dir: " ${PRIMITIV_C_INCLUDE_DIR})
-  #message("-- Primitiv C API libraries: " ${PRIMITIV_C_LIBRARIES})
+  message("-- Found primitiv C API: " ${PRIMITIV_C_LIBRARIES})
+  #message("-- primitiv C API include dir: " ${PRIMITIV_C_INCLUDE_DIR})
+  #message("-- primitiv C API libraries: " ${PRIMITIV_C_LIBRARIES})
   set(PRIMITIV_C_FOUND ON)
 else()
   message(STATUS "Primitiv C API is not installed.")
```
Comment (vbkaisetsu) [inline cmake/FindPrimitiv.cmake]: `Primitiv` → `primitiv`?

**Review by vbkaisetsu — DISMISSED** [review]
`Primitiv` → `primitiv`
Please see a comment.

File: cmake/FindPrimitiv.cmake [hunk 2]
```diff
@@ -77,12 +81,58 @@ find_library(PRIMITIV_C_LIBRARIES NAMES primitiv_c
...
+
+string(REGEX MATCH
+  "#[ \t]*define[ \t]+PRIMITIV_VERSION_PATCH[ \t]+([0-9]+)"
+  _primitiv_version_match "${_primitiv_version_header}"
+)
+set(PRIMITIV_VERSION_PATCH "${CMAKE_MATCH_1}")
+
+set(PRIMITIV_VERSION "${PRIMITIV_VERSION_MAJOR}.${PRIMITIV_VERSION_MINOR}.${PRIMITIV_VERSION_PATCH}")
+
+#message("primitiv version requested: ${PRIMITIV_FIND_VERSION}")
```
Comment (vbkaisetsu) [inline cmake/FindPrimitiv.cmake]: `PRIMITIV_FIND_VERSION` → `Primitiv_FIND_VERSION`?

Comment (vbkaisetsu) [inline cmake/FindPrimitiv.cmake]: `PRIMITIV_FIND_VERSION` → `Primitiv_FIND_VERSION`?

File: cmake/FindPrimitiv.cmake (hunk 1 above)
Comment (odashi) [reply cmake/FindPrimitiv.cmake]: Fixed.

File: cmake/FindPrimitiv.cmake (hunk 2 above)
Comment (odashi) [reply cmake/FindPrimitiv.cmake]: Fixed.

**Review by chantera — APPROVED** [review]
LGTM.

**Pull request merged**